### ISSUE TITLE
dict-items-not-iterating false positive when passed to [].extend

### ISSUE THREAD

**Issue by PCManticore** [PyCQA/pylint#2187]
### Steps to reproduce
Run pylint over the following file:
```
a = {}
[].extend(a.items())
``` 

### Current behavior

We're emitting `dict-items-not-iterating`,

### Expected behavior

We shouldn't emit that as the `extend()` function call is an iterating context, we probably need to update the `ACCEPTS_ITERATOR` list to have `extend` as well.

### pylint --version output
1.9.2, didn't try with `master`.

**Comment by brycepg** [comment]
I can't reproduce this with

    pylint 1.9.2,
    astroid 1.6.5
    Python 3.6.5rc1 (v3.6.5rc1:f03c5148cf, Mar 14 2018, 03:12:11) [MSC v.1913 64 bit (AMD64)]

**Comment by PCManticore** [comment]
It reproduces for me:
```
************* Module a
W:  2,10: dict.items referenced when not iterating (dict-items-not-iterating)
```
Forgot to mention that this is with the Python 3 porting checker, with `--py3k`.
